src/compiler/checker.ts

```diff
@@ -524,6 +524,9 @@ namespace ts {
             },
             getApparentType,
             getUnionType,
+            isTypeAssignableTo: (source, target) => {

```

Comment:
Even `@internal`, some people are going to be ecstatic that this finally found its way into our API surface and will therefore finally be callable by people using the API (even if via cast).

Reply:
I'm really interested in the reason why it should be marked as `@internal`. Don't get me wrong, I'm one of those people ecstatic that this found it's way to the API, now I can ditch my [custom typescript version](https://github.com/SamVerschueren/tsd/tree/master/libraries/typescript) from my library which exposed this. I'm just super curious about the reasoning behind this decision :)

Reply:
Effectively this function now has to cross conceptual boundaries in TypeScript. Going from being useful internally only inside checker to being used inside the TSServices, which meant it needed to be exposed as a public API on the checker internally within our codebase,

Reply:
Ok, but why the `@internal` annotation? It looks like the method is still kinda private?

Reply:
I'm not 100% convinced that arbitrary uses of this function are safe and won't corrupt compiler state. Marking it `@internal` allows me to use it inside Typescript where I know how it's being used. Using it outside is an `any` cast away, but that way it's obvious that it's not technically supported.